src/ProductConstructionService/ProductConstructionService.Api/Queue/JobsProcessor.cs

```diff
@@ -24,6 +29,15 @@ protected override async Task ExecuteAsync(CancellationToken cancellationToken)
         // The API won't be initialized until the BackgroundService goes async. Since the scopeManagers blocks aren't async, we have to do it here
         await Task.Delay(1000);
 
+        if (_options.Value.CloneVmr)
+        {
+            using (ITelemetryScope scope = telemetryRecorder.RecordGitClone(Constants.DefaultVmrUri))
+            {
+                await vmrCloner.PrepareVmrCloneAsync(cancellationToken);
+                scope.SetSuccess();
+            }
+        }
```

Comment:
This does not belong here. Can you add it as a startup filter?

Reply:
Yeah I'll do that. I forgot that we'll have the old API listening and queuing items until this one is up and ready, so I put it there to not process any items until the VMR is ready

Reply:
I implemented it as a `StartupFilter`, now I need to dial in the Startup Health check to wait for the VMR checkout, hopefully the 60sec max time delay won't be an issue

Reply:
What do you mean	`60sec max time delay`?

Reply:
There's a configuration parameter or the Startup Health check, `InitialDelaySeconds`, which tells the ContainerApp how long after the startup it should run the health check.

Reply:
But doesn't it then run it repeatedly until it becomes healthy?
What's the life cycle?

I am not sure the clone will happen within 60 seconds

Reply:
It will wait 60 seconds to start. Then it will attempt to reach the health endpoint, until it succeeds once, or fails `failureThreshold` times. Max value for `failureThreshold` is 10

Reply:
there is no parameter to configure this, so I guess the attempts are sequential

Reply:
We have a startup test that is supposed to test if the service has started successfully.
The health endpoint we have just tests if the service is responsive, however, since the endpoint won't be initialized until the VMR is initialized, it kind of acts like a readiness endpoint in our case

Reply:
I'm doing some testing now, we might be able to use the `timeoutSeconds` parameter to increase the time we can wait before failing

Reply:
No luck, maybe we should just write our own health probes in the deployment? We were only using the Startup one, so it should be quite easy

Reply:
Yeah, possibly. The documentation looks quite simple?

Reply:
I'd just use the `/health` endpoint we already have. The only change would be in the deployment script itself, it would wait, maybe 2 min, before trying to get a response from the health endpoint, and then we'd have the same logic as the integrated health probe. Shame that they put such a low initial delay. Maybe we can ask to get it increased

Reply:
And these are defaults of the ACA probe?

Reply:
So we could increase the `failureThreshold`, `periodSeconds` and `initialDelaySeconds` to give it enough time?